generator/src/lib.rs

```diff
@@ -17,6 +17,21 @@ use std::fmt::Display;
 use std::path::Path;
 use syn::Ident;
 
+macro_rules! maybe_extract {
```

Comment:
Would `get_variant` be a clearer name for this?

Reply:
I wasn't sure what name was clearest and had tried a bunch of different ones... `get_variant` sounds good to me; I'll switch to that.